From ae8550a860a40801041f348d1b85b134c894687b Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Tue, 28 Jun 2022 17:39:15 +0800
Subject: [PATCH 14/15] arm64: allwinner: dts: pinetab: add BMA223
 accelerometer

PineTab has an BMA223 accelerometer that is mounted with a different
orientation than the LCD panel.

Add it to the device tree with an mount matrix that rotating the
orientation.

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 arch/arm64/boot/dts/allwinner/sun50i-a64-pinetab.dts | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinetab.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinetab.dts
index 5d8081a093b6..a4f12f79f5a5 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-pinetab.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-pinetab.dts
@@ -237,7 +237,15 @@ &i2c0_pins {
 &i2c1 {
 	status = "okay";
 
-	/* TODO: add Bochs BMA223 accelerometer here */
+	bma223@18 {
+		compatible = "bosch,bma223";
+		reg = <0x18>;
+		interrupt-parent = <&pio>;
+		interrupts = <7 5 IRQ_TYPE_LEVEL_HIGH>; /* PH5 */
+		mount-matrix = "0", "-1", "0",
+			       "-1", "0", "0",
+			       "0", "0", "-1";
+	};
 };
 
 &lradc {
-- 
2.36.0

